spec/std/dir_spec.cr

```diff
@@ -10,223 +10,228 @@ end
       ].sort
     end
 
     it "tests with ** (same as *)" do
-      Dir["#{__DIR__}/data/dir/**"].sort.should eq [
-        File.join(__DIR__, "data", "dir", "dots"),
-        File.join(__DIR__, "data", "dir", "f1.txt"),
-        File.join(__DIR__, "data", "dir", "f2.txt"),
-        File.join(__DIR__, "data", "dir", "f3.txx"),
-        File.join(__DIR__, "data", "dir", "g2.txt"),
-        File.join(__DIR__, "data", "dir", "subdir"),
-        File.join(__DIR__, "data", "dir", "subdir2"),
+      Dir["#{datapath}/dir/**"].sort.should eq [
+        datapath("dir", "dots"),
+        datapath("dir", "f1.txt"),
+        datapath("dir", "f2.txt"),
+        datapath("dir", "f3.txx"),
+        datapath("dir", "g2.txt"),
+        datapath("dir", "subdir"),
+        datapath("dir", "subdir2"),
       ].sort
     end
 
     it "tests with */" do
-      Dir["#{__DIR__}/data/dir/*/"].sort.should eq [
-        File.join(__DIR__, "data", "dir", "dots", ""),
-        File.join(__DIR__, "data", "dir", "subdir", ""),
-        File.join(__DIR__, "data", "dir", "subdir2", ""),
+      Dir["#{datapath}/dir/*/"].sort.should eq [
+        datapath("dir", "dots", ""),
+        datapath("dir", "subdir", ""),
+        datapath("dir", "subdir2", ""),
       ].sort
     end
 
     it "tests glob with a single pattern with extra slashes" do
-      Dir["#{__DIR__}////data////dir////*.txt"].sort.should eq [
-        File.join(__DIR__, "data", "dir", "f1.txt"),
-        File.join(__DIR__, "data", "dir", "f2.txt"),
-        File.join(__DIR__, "data", "dir", "g2.txt"),
+      Dir["spec/std////data////dir////*.txt"].sort.should eq [
+        datapath("dir", "f1.txt"),
+        datapath("dir", "f2.txt"),
+        datapath("dir", "g2.txt"),
       ].sort
     end
 
     it "tests with relative path" do
-      Dir["spec/std/data/dir/*/"].sort.should eq [
-        File.join("spec", "std", "data", "dir", "dots", ""),
-        File.join("spec", "std", "data", "dir", "subdir", ""),
-        File.join("spec", "std", "data", "dir", "subdir2", ""),
+      Dir["#{datapath}/dir/*/"].sort.should eq [
+        datapath("dir", "dots", ""),
+        datapath("dir", "subdir", ""),
+        datapath("dir", "subdir2", ""),
       ].sort
     end
 
     it "tests with relative path (starts with .)" do
-      Dir["./spec/std/data/dir/*/"].sort.should eq [
+      Dir["./#{datapath}/dir/*/"].sort.should eq [
         File.join(".", "spec", "std", "data", "dir", "dots", ""),
         File.join(".", "spec", "std", "data", "dir", "subdir", ""),
         File.join(".", "spec", "std", "data", "dir", "subdir2", ""),
       ].sort
     end
 
     it "tests with relative path (starts with ..)" do
-      base_path = File.join("..", File.basename(File.dirname(File.dirname(__DIR__))), "spec", "std", "data", "dir")
-      Dir["../#{File.basename(File.dirname(File.dirname(__DIR__)))}/spec/std/data/dir/*/"].sort.should eq [
+      base_path = File.join("..", File.basename(Dir.current), "spec", "std", "data", "dir")
```

Comment:
I would think that relaying on `__DIR__` is better that `Dir.current` since the later involves more state.
Why this change? 
I also notice that the `datapath` helper relies on the current directory and not an absolute path by using `__DIR__`

Reply:
`__DIR__` is compile time, `Dir.current` runtime. In order for the specs to correctly execute when cross-compiled, we need to use `Dir.current` because `__DIR__` might not even exist on the target system.

Reply:
Ok, understood.